src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars SET model = ?, manufacturer_id = ? WHERE id = ? "
+                + "AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement = connection
+                        .prepareStatement(updateCarQuery, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.setLong(3, car.getId());
+            createCarStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", throwable);
+        }
+        clearDriverCarRelations(car);
+        List<Driver> drivers = car.getDrivers();
+        if (drivers != null) {
+            for (Driver driver : drivers) {
+                setDriverCarRelation(driver, car);
+            }
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement = connection
+                        .prepareStatement(deleteCarQuery)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getCarByDriver(Long driverId) {
+        String findCarsQuery = "SELECT c.id AS car_id, model, manufacturer_id "
+                + " FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id WHERE "
+                + "cd.driver_id = ? AND c.is_deleted = FALSE;";
+        List<Car> allCars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsStatement = connection.prepareStatement(findCarsQuery)) {
+            getCarsStatement.setLong(1, driverId);
+            ResultSet resultSet = getCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                allCars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find cars by driver id: " + driverId, e);
+        }
+        return allCars;
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject(1, Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        Manufacturer manufacturer = manufacturerDao.get(manufacturerId).get();
```

Comment:
that's a bad practice open connection inside of another connection. 
Let's map manufacturer manually, it will force us to have code duplication but in this case it's better